crates/sui-bridge-indexer/src/sui_transaction_handler.rs

```diff
@@ -34,7 +36,8 @@ pub async fn handle_sui_transactions_loop(
     let mut stream = ReceiverStream::new(rx).ready_chunks(checkpoint_commit_batch_size);
     while let Some(batch) = stream.next().await {
         // unwrap: batch must not be empty
-        let cursor = batch.last().unwrap().1;
+        let (txns, cursor) = batch.last().cloned().unwrap();
+        let last_ckp = txns.last().map(|tx| tx.checkpoint).unwrap_or_default();
```

Comment:
why is this line moved here?

Reply:
this was probably needed for some other reason previously, will revert it.